### ISSUE TITLE
Failed to start OPC-UA Connection

### ISSUE THREAD

**Issue by chenkel** [aws-quickstart/quickstart-aws-industrial-machine-connectivity#49]
### Problem: 
I receive the MQTT messages on `spBv1.0/AWS Smart Factory/DDATA/Smart Factory 1` but the SiteWise connection does not seem to work. There are no _Latest Values_ showing up as explained in the [post-deployment](https://github.com/aws-quickstart/quickstart-aws-industrial-machine-connectivity/blob/main/assets/readme/post-deployment.md#validate-incoming-data).

### What did I do?
I set up the Data Flow Option 1 (OPC UA via SiteWise Connector) in us-east-1 and eu-central-1. 
For the AMC driver I chose IgnitionCirrusLink.
In both region, I followed all steps including the Post-Deployment steps and the Ignition software configuration steps.
It is in a fresh AWS environemnt and I did not do any further customizations beyond the instructions.

### What do I observe?
Every few seconds in the Log Group `/aws/greengrass/Lambda/us-east-1/aws/swCollector` in CloudWatch Logs:

> WARN OpcUaManagedConnection:132
> Failed to start OPC-UA Connection for Source Automated Gateway Config
> com.amazon.iot.bifrost.opcua.exception.CertificateStoreException: Failed to add key to store 

In Ignition Logs I find:

> com.cirruslink.chariot.client.CirrusClient: VirtualDevice: Connected to ssl://[I-Deleted-That]-ats.iot.us-east-1.amazonaws.com:8883

But after that:

> com.cirruslink.mqtt.common.gateway.agent.AgentEdgeNode
> Error attempting to convert folder _ types _ [spaces due to Markdown formatting]
> java.lang.NullPointerException: null

### Question
What am I doing wrong?

I will test a different AMC driver now...

**Comment by leovac** [comment]
@chenkel Had a similar issue and  the problem is with wrong java version. 
Here the fix: 
`apt-cache madison openjdk-8-jdk
wget https://download.bell-sw.com/java/8u282+8/bellsoft-jdk8u282+8-linux-amd64.deb
apt install ./bellsoft-jdk8u282+8-linux-amd64.deb
 `
 And then check if java  version is 282.

**Comment by asvand** [comment]
Having the same issue, @chenkel  was it resolved on your side?

**Comment by chenkel** [comment]
No, I ran out of time and had to use the working resources from the [AWS IoT SiteWise Workshop](https://iot-sitewise.workshop.aws/en/aws-iot-sitewise-monitor.html), which were an even better fit for my use case.

Luka's workaround might fix it, but I haven't tried it.

**Comment by robertsarkozi** [comment]
I'm experiencing the same error message in our QuickStart package.

Someone else has reported this on their forums as well: https://forums.aws.amazon.com/message.jspa?messageID=991038

Has anyone tried @leovac's suggestion on how to fix this?

**Comment by YTvW** [comment]
@robertsarkozi 

That forum post is the one I made, I have encountered this problem now on multiple machines and the fix that @leovac suggested has worked for me on all of them.
You just need to figure out which architecture specific version you need to download

**Comment by robertsarkozi** [comment]
Thanks @YTvW.

I did install the right version, but it didn't work. Java executables couldn't be found by the SiteWise components after installing it. I'm using an amazonlinux AMI for the EC2 instance I'm running everything on...

Should we be the ones to fix this actually? Something clearly is broken because my original setup didn't change in the package I'm maintaining, so something must have changed in the AWS backend...

**Comment by YTvW** [comment]
No problem @robertsarkozi 

Can you check if the there is a file or symlink to java with the name java8 in the /usr/bin folder? since aws build greengrass in such a way that it needs to have that exact name for it to recognise is as the correct executable

**Comment by robertsarkozi** [comment]
Can anyone confirm that this has been fixed? My package started working again, without any sign of the error message.

**Comment by witussk** [comment]
> Can anyone confirm that this has been fixed? My package started working again, without any sign of the error message.

I can say for sure that under my configuration it did not work out of the box using the default open-jdk. But the solution to install a specific version of Java as @leovac suggested did work for me for 2 different devices.
